flang/test/Semantics/OpenMP/declarative-directive.f90

```diff
@@ -23,6 +23,7 @@ end subroutine requires_2
 
 subroutine declare_simd_1(a, b)
   real(8), intent(inout) :: a, b
+  !ERROR: 'a' in ALIGNED clause must be of type C_PTR, POINTER or ALLOCATABLE
```

Comment:
Do we also deal with ALIGNED clause in this PR? A similar issue is causing the Linux CI failure:

```
flang/test/Examples/omp-declarative-directive.f90:11:38: error: 'a' in ALIGNED clause must be of type C_PTR, POINTER or ALLOCATABLE
--
  | !$omp declare simd(declare_simd_1) aligned(a)
```

Reply:
Yes, this line: https://github.com/llvm/llvm-project/pull/111354/files#diff-be225bc038e95a229024c9f227284112f98fe31762bc0bb80605de425aadd084R2287 is responsible for the change.

Reply:
I fixed it in the recent commit.